src/Role/Service/FetchUserPermissionService.php

```diff
@@ -34,24 +34,21 @@ public function __construct(
      *
      * @return CustomResponseInterface
      */
-    public function execute(int $userId): CustomResponseInterface
+    public function execute(int $rank): CustomResponseInterface
     {
-        /** @var array $userRoleIds */
-        $userRoleIds = $this->roleUserRepository->getUserRoleIds($userId);
-
-        if (!$userRoleIds) {
-            return response()
-                ->setData([]);
+        /** @var array $rankRoleIds */
+        $rankRoleIds = $this->roleRankRepository->getRankRoleIds($rank);
+        
+        if (!$rankRoleIds) {
+            return response()->setData([]);
```

Comment:
We could probably throw an error here? Is it needed to return with an empty array? Maybe respond with a custom error code for the frontend?

Are there ranks without roles?

Reply:
Well, rank 1 would be a rank without role

Reply:
And there would be also Roles without Ranks just as "Root" Level or "Group" level Roles